### PR TITLE
HTTP Web server and gcp

### PR DESCRIPTION
**Pull request by rtrive**

### CODE REVIEW

File: .github/workflows/ci-cd.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+on: push
+
+name: Build and Deploy a Container
+env:
+  PROJECT_ID: ${{ secrets.GCP_PROJECT }}
+  SERVICE: f1-telegram-bot
+  REGION: europe-west4
+  BUCKET: test-riccardo
```
Comment (kin0992) [inline .github/workflows/ci-cd.yaml:8]: Is this the right value?

File: src/http_server.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class PythonServer(SimpleHTTPRequestHandler):
+    def do_GET(self):
+        logging.info("request GET / ")
+        self.send_response(200)
+        self.send_header("Content-type", "text/html")
+        self.end_headers()
+        self.wfile.write("GET request for {}".format(self.path).encode("utf-8"))
+
+
+def start():
```
Comment (kin0992) [inline src/http_server.py:20]:
```suggestion
def start(logger: Logger):
```
and pass the logger to this function from the `main.py` file (as we have done already)

File: src/main.py [hunk 3]
```diff
@@ -51,6 +51,8 @@ def get_player_cookie(log: Logger, driver: uc_chrome) -> str:
     scheduler.add_job(func=reboot, trigger="interval", hours=24, kwargs={"log": log})
     scheduler.start()
 
+    http_server_start()
```
Comment (kin0992) [inline src/main.py:54]:
```suggestion
    http_server_start(logger=create_logger(
        name="http-server", level=configuration.log.log_level, format=LOG_FORMAT
    ))
```

File: src/http_server.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import logging
+from http.server import SimpleHTTPRequestHandler, ThreadingHTTPServer
+from threading import Thread
+
+logger = logging.getLogger(__name__)
+
+PORT = 8080
+HOSTNAME = "localhost"
```
Comment (kin0992) [inline src/http_server.py:8]: Should we move these variables to environment variables?

And then create a separate configuration within the `configuration.py`, something like `HttpServerConfiguration`. What do you think?

**Review by kin0992 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-cd.yaml (hunk 1 above)
Comment (rtrive) [reply .github/workflows/ci-cd.yaml:8]: yep, but i added others right now

File: src/main.py [hunk 5]
```diff
@@ -30,6 +30,11 @@ def get_player_cookie(log: Logger, driver: uc_chrome) -> str:
     player_cookie = ""
     log.debug("get cookie")
     try:
+        request = driver.wait_for_request(
+            "/v2/account/subscriber/authenticate/by-password", 120
+        )
+        log.info(request.body)
+        log.info(request.response.body)
```
Comment (kin0992) [inline src/main.py:37]:
```suggestion
```

File: src/main.py [hunk 6]
```diff
@@ -29,17 +30,26 @@ def get_player_cookie(log: Logger, driver: uc_chrome) -> str:
...
+        reboot(log)
     return player_cookie
 
 
 if __name__ == "__main__":
-    load_dotenv()
+    if not os.environ.get("RUNNING_IN_DOCKER"):
+        print("load from .env file")
+        load_dotenv()
+
```
Comment (kin0992) [inline src/main.py:52]: After what we have discovered, I'm not sure this is needed

**Review by kin0992 — APPROVED** [review]

**Pull request merged**